conans/assets/templates/new_v2_bazel.py

```diff
@@ -24,28 +26,28 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
-    def validate(self):
-        if self.settings.os in ("Windows", "Macos") and self.options.shared:
-            raise ConanException("Windows and Macos needs extra BUILD configuration to be able "
-                                 "to create a shared library. Please, check this reference to "
-                                 "know more about it: https://bazel.build/reference/be/c-cpp")
-
     def layout(self):
-        bazel_layout(self)
         # DEPRECATED: Default generators folder will be "conan" in Conan 2.x
         self.folders.generators = "conan"
+        bazel_layout(self)
 
     def build(self):
         bazel = Bazel(self)
-        bazel.build()
+        # Linux creates both shared and static libraries by default, and
+        # it is even conflicting if we use the cc_shared_library rule
```

Comment:
I tried with the `cc_shared_library` and apparently it worked, could you re-check?

Reply:
False alarm, trying again, indeed it fails because both rules try to create the same .so file in Linux, leading to a conflict.

Reply:
For the record, the error that is showing up is:
```
...
PrimaryInput: File:[[<execution_root>]bazel-out/k8-opt/bin]main/_objs/hello/hello.pic.o
PrimaryOutput: File:[[<execution_root>]bazel-out/k8-opt/bin]main/libhello.so
Owner information: ConfiguredTargetKey{label=//main:hello, config=BuildConfigurationKey[21071ba4bd5859dd8165bf6341a4d915b0b94f2332e31389877d71fded5146ab]}, ConfiguredTargetKey{label=//main:hello_shared, config=BuildConfigurationKey[21071ba4bd5859dd8165bf6341a4d915b0b94f2332e31389877d71fded5146ab]}
MandatoryInputs: are equal
Outputs: are equal
ERROR: com.google.devtools.build.lib.actions.MutableActionGraph$ActionConflictException: for main/libhello.so-2.params, previous action: action 'Writing file main/libhello.so-2.params', attempted action: action 'Writing file main/libhello.so-2.params'
INFO: Elapsed time: 3.210s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (36 packages loaded, 158 targets configured)
```
It seems a name collision. I faced it when I ran a `conan create .` which creates a `.so` and a `.a` on Linux, and then I ran a `conan create . -o *:shared=True`.